### PR TITLE
Adding controls to dlgPolarCluster

### LINKED ISSUE #5966: More circular plots from openair package
There are 6 more circular plots provided by the Openair package.
Percentile Rose
Frequency Rose
Polar Plot
Polar Cluster
Polar Annulus
Polar Frequency
These may become 6 dialogues or (say) 2 dialogues with 3 radio buttons in each or...?  From a quick look I fail to see the connections that would help in structuring the dialogues.  Once decided they should be easy to construct and code, because they have similar options to the Wind/Pollution Rose function.

This is work initially for @HawardKetoyoMsatsi with @Wycklife probably doing the coding then.

### PR DESCRIPTION
**Pull request by HawardKetoyoMsatsi**
Fixes #5966
Added controls to dlgPolarCluster and implemented R code.

@africanmathsinitiative/developers , @shadrackkibet , this is ready for review.

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
I can't test this until #6048 is merged.

File: instat/instat.vbproj.user [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
     <FallbackCulture>en-US</FallbackCulture>
     <VerifyUploadedFiles>false</VerifyUploadedFiles>
-    <ProjectView>ShowAllFiles</ProjectView>
+    <ProjectView>ProjectFiles</ProjectView>
   </PropertyGroup>
   <PropertyGroup>
     <EnableSecurityDebugging>false</EnableSecurityDebugging>
   </PropertyGroup>
-</Project>
+</Project>
```
Comment (shadrackkibet) [inline instat/instat.vbproj.user:17]: Did you intend to have these changes here?

File: instat/dlgPolarCluster.vb [hunk 2]
```diff
@@ -1,3 +1,117 @@
...
+    Private Sub SetRCodeForControls(bReset As Boolean)
+        SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, bReset)
+    End Sub
+
+    Private Sub TestOkEnabled()
+        If ucrReceiverPollutant.IsEmpty Then
+            ucrBase.OKEnabled(False)
+        Else
+            ucrBase.OKEnabled(True)
+        End If
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:105]: Just to check, what if the numeric up-downs are empty? doesn't this bug?

File: instat/dlgPolarCluster.vb [hunk 3]
```diff
@@ -1,3 +1,117 @@
...
+        ucrInputColor.SetParameter(New RParameter("cols", 4))
+        dctColor.Add("Accent", Chr(34) & "Accent" & Chr(34))
+        dctColor.Add("Dark2", Chr(34) & "Dark2" & Chr(34))
+        dctColor.Add("Paired", Chr(34) & "Paired" & Chr(34))
+        dctColor.Add("Pastel1", Chr(34) & "Pastel1" & Chr(34))
+        dctColor.Add("Pastel2", Chr(34) & "Pastel2" & Chr(34))
+        dctColor.Add("Set1", Chr(34) & "Set1" & Chr(34))
+        dctColor.Add("Set2", Chr(34) & "Set2" & Chr(34))
+        dctColor.Add("Set3", Chr(34) & "Set3" & Chr(34))
+        ucrInputColor.SetItems(dctColor)
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:69]: set the R Default here

**Review by shadrackkibet — COMMENTED** [review]
Looks good. Then think about data at multiple levels, will this function still cope with that?

**Comment by shadrackkibet** [comment]
I think you also want to save the graph so add the save control into these dialogs.

File: instat/dlgPolarCluster.vb (hunk 2 above)
Comment (HawardKetoyoMsatsi) [reply instat/dlgPolarCluster.vb:105]: It doesnt bug however the code appears as follows.

last_graph <- openair::polarCluster(mydata=mydata, pollutant="nox", wd="wd", x="ws", cols="Accent", angle=, n.clusters=)

should we leave the OK button disabled until these fields are filled?

File: instat/dlgPolarCluster.vb (hunk 3 above)
Comment (HawardKetoyoMsatsi) [reply instat/dlgPolarCluster.vb:69]: @shadrackkibet the default is already set to "Accent".

Comment (shadrackkibet) [reply instat/dlgPolarCluster.vb:69]: But that is not the R Default

File: instat/dlgPolarCluster.vb (hunk 2 above)
Comment (shadrackkibet) [reply instat/dlgPolarCluster.vb:105]: So long as it doesn't bug and the graph produced is sensible then happy to leave it that way.

**Comment by lloyddewit** [comment]
@rdstern PR #6048 is now merged so this PR is ready for testing (but you may wish to wait for @HawardKetoyoMsatsi to resolve @shadrackkibet 's suggestions first)

File: instat/instat.vbproj.user (hunk 1 above)
Comment (HawardKetoyoMsatsi) [reply instat/instat.vbproj.user:17]: @shadrackkibet No, I didn't intent to have them. 
@lloyddewit I am not able to resolve the changes made in the file **instat/instat.vbproj.user** I didn't intend to have them is there anything that can be done about it?

Comment (lloyddewit) [reply instat/instat.vbproj.user:17]: @HawardKetoyoMsatsi I think you can just reverse the changes to `instat.vbproj.user` and push the file again (i.e. copy-paste the original contents of the file into your version of the file and push it again). Git will recognise that the file is identical and remove it automatically from the PR (I think!)

**Comment by lloyddewit** [comment]
@HawardKetoyoMsatsi are the comments from @shadrackkibet now resolved? thanks

**Comment by HawardKetoyoMsatsi** [comment]
@lloyddewit in the last 7 commits I made in this PR I addressed comments made by @shadrackkibet.

**Comment by lloyddewit** [comment]
@rdstern please could you test? thanks

**Comment by lloyddewit** [comment]
@rdstern please could you test? thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
@HawardKetoyoMsatsi  well done this all seems to work.

I am not sure what it does, so it will be good to get it merged soon.  I have a few questions:
a) The first is called pollutant.  That's odd to me that it isn't a default of wind speed?
b) There is an odd order of the controls with the Date between the other two.
c) You have to include a third variable.  I found that I could put windspeed as the pollutant and also the third variable.  So I am still not quite sure what is happening here.
d) This probably applies generally to this set of controls.  None of them seem to fill by default.  If the data were defined as climatic with these variables, then perhaps they could be?

**Comment by HawardKetoyoMsatsi** [comment]
I am closing this so that I can reimplement and open it in another PR

**Pull request closed without merging**